darts/models/forecasting/baselines.py

```diff
@@ -200,13 +209,14 @@ def fit(
             past_covariates=past_covariates,
             future_covariates=future_covariates,
         )
-        for model in self.models:
-            kwargs = dict(series=series)
-            if model.supports_past_covariates:
-                kwargs["past_covariates"] = past_covariates
-            if model.supports_future_covariates:
-                kwargs["future_covariates"] = future_covariates
-            model.fit(**kwargs)
+        if self.retrain_forecasting_models:
+            for model in self.forecasting_models:
+                kwargs = dict(series=series)
+                if model.supports_past_covariates:
+                    kwargs["past_covariates"] = past_covariates
+                if model.supports_future_covariates:
+                    kwargs["future_covariates"] = future_covariates
+                model.fit(**kwargs)
```

Comment:
here the tip again to return before in order to avoid indented sections

```suggestion
        if not self.retrain_forecasting_models:
            return self

        for model in self.forecasting_models:
            kwargs = dict(series=series)
            if model.supports_past_covariates:
                kwargs["past_covariates"] = past_covariates
            if model.supports_future_covariates:
                kwargs["future_covariates"] = future_covariates
            model.fit(**kwargs)
```